tierkreis_visualization/tierkreis_visualization/data/models.py

```diff
@@ -1,23 +1,25 @@
 from typing import Literal, Any
 from pydantic import BaseModel
+from tierkreis.controller.data.location import Loc
 
 NodeStatus = Literal["Not started", "Started", "Error", "Finished"]
 
 
 class PyNode(BaseModel):
-    id: int
+    id: Loc
     status: NodeStatus
     function_name: str
+    node_type: str
```

Comment:
Similar to NodeStatus you could literally type the union and remove casting in the frontend for a stricter contract.
```suggestion
    node_type: Literal["eval", "loop", "map"]
```